app/controllers/api/profile_controller.rb

```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-class Api::LocaleController < ApiController
+class Api::ProfileController < ApiController
```

Comment:
Try to crudify this controller:
- remove update method
- override entry method to just return current_user

That should already do the trick :)